### PR TITLE
Port WebBrowser completion events fix from 4.8

### PR DESCRIPTION
**Pull request by SamBent**
Addresses Issue #4990

## Description

When WebBrowser navigates to about:blank, it cancels outstanding navigations.  However it may still receive completion events for those navigations, due to race conditions between the browser process and the WPF app.  When this happens, ignore the completion event (instead of failing an assert and crashing).

## Customer Impact

Intermittent crashes.

## Regression

No.

## Testing

Ad-hoc around customer scenario.
Standard regressions testing.

## Risk

Low.  Port of .NETFx servicing fix released earlier this year.

### CODE REVIEW

**Review by ryalanms — APPROVED** [review]

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Controls/WebBrowserEvent.cs [hunk 1]
```diff
@@ -248,14 +248,23 @@ public void DocumentComplete(object pDisp, ref object url)
             }
         }
 
+        // if we are navigating to "about:blank", ignore completion events
+        // for any other url.  These can happen despite our attempt to cancel
+        // outstanding navigations (see WebBrowser.DoNavigate), due to race
+        // conditions between the browser and us.
+        private bool ShouldIgnoreCompletionEvent(ref object url)
```
Comment (batzen) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Controls/WebBrowserEvent.cs:255]: Do we really need ref here?

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Controls/WebBrowserEvent.cs [hunk 2]
```diff
@@ -248,14 +248,23 @@ public void DocumentComplete(object pDisp, ref object url)
...
+            string urlString = url as string;
+            return (_parent.NavigatingToAboutBlank &&
+                    String.Compare(urlString, WebBrowser.AboutBlankUriString, StringComparison.OrdinalIgnoreCase) != 0);
+        }
+
         public void CommandStateChange(long command, bool enable)
         {
-            if (command == NativeMethods.CSC_NAVIGATEBACK)
-            {
+            if (command == NativeMethods.CSC_NAVIGATEBACK) {
```
Comment (batzen) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Controls/WebBrowserEvent.cs:264]: Shouldn't the line be wrapped before the curly brace?

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Controls/WebBrowserEvent.cs (hunk 1 above)
Comment (SamBent) [reply src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Controls/WebBrowserEvent.cs:255]: Maybe not, but copying the conventions of existing code.  There's no gain in changing it.

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Controls/WebBrowserEvent.cs (hunk 2 above)
Comment (SamBent) [reply src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Controls/WebBrowserEvent.cs:264]: Yes.  This got changed during the original port from .NET 4.8 to .NET Core 3.0, and accidentally changed back again when porting this fix, by a tool (or tool operator) who didn't notice.

**Pull request merged**